### PR TITLE
replace uses of deprecated `std::iter::{range, range_step}`

### PR DESCRIPTION
**Pull request by apasel422**

### CODE REVIEW

**Comment by FlashCat** [comment]
r? @cgaebel

(I've picked a reviewer for you, use r? to override)

File: src/map.rs [hunk 1]
```diff
@@ -1269,31 +1268,31 @@ mod test {
         let mut trie = Map::new();
         let n = 300;
 
-        for x in range_step(1, n, 2) {
+        for x in (1..n).step_by(2) {
```
Comment (Gankra) [inline src/map.rs:1271]: :O didn't know this had landed

**Comment by Gankra** [comment]
@FlashCat r+

**Comment by FlashCat** [comment]
:pushpin: Commit 94f0acd has been approved by `Gankro`

**Comment by FlashCat** [comment]
:hourglass: Testing commit 94f0acd with merge f0fefc2...

**Comment by FlashCat** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/contain-rs/trie/builds/55177699)

**Pull request merged**